### PR TITLE
[EGv2] generate with newer emitter

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py [hunk 1]
```diff
@@ -89,21 +98,33 @@ async def _publish_cloud_event(  # pylint: disable=inconsistent-return-statement
...
 
     @distributed_trace_async
-    async def _publish_cloud_events(  # pylint: disable=inconsistent-return-statements
+    async def _publish_cloud_events(  # pylint: disable=protected-access
         self,
         topic_name: str,
         events: List[_models._models.CloudEvent],
         **kwargs: Any
-    ) -> None:
+    ) -> _models._models.PublishResult:
```
Comment (kashifkhan) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py:127]: is the return type change here going to impact anything public facing ?

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py [hunk 2]
```diff
@@ -132,11 +153,11 @@ async def _publish_cloud_events(  # pylint: disable=inconsistent-return-statemen
         _params = kwargs.pop("params", {}) or {}
 
         content_type: str = kwargs.pop('content_type', _headers.pop('content-type', "application/cloudevents-batch+json; charset=utf-8"))
-        cls: ClsType[None] = kwargs.pop(
+        cls: ClsType[_models._models.PublishResult] = kwargs.pop(  # pylint: disable=protected-access
             'cls', None
         )
 
-        _content = json.dumps(events, cls=AzureJSONEncoder)  # type: ignore
+        _content = json.dumps(events, cls=AzureJSONEncoder, exclude_readonly=True)  # type: ignore
```
Comment (kashifkhan) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py:160]: just double checking, has the default behavior always been to exclude readonly ?

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py (hunk 1 above)
Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py:127]: Nope this function is private:)

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py (hunk 2 above)
Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py:160]: I believe so -- this was done by the code gen

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**